### PR TITLE
[IMP] open_academy: model-constraints t#59945

### PR DESCRIPTION
**Pull request by JJRR5**
### **This PR is reference to [T#59945](https://www.vauxoo.com/web#action=464&active_id=17&cids=1&id=59945&menu_id=1490&model=project.task&view_type=form)**

**Issue**
The main goal of this section was to understand how to create constraints with python and SQL.

**What changed**

- Created python constraints that checks that the instructor is not present in the attendees of his/her own session.
- Added SQL constraints : 
    created CHECK constraint to ensure that the course description and the course title are different
    created a constraint that made the Course’s name's UNIQUE
- Re-implemented the copy method

**Evidence**

**Instructor cannot be an attendee of his own session**
image

**Description and title must be different in a Course**
image

**Unique Course's name**
image

**Copy Method**
image

### CODE REVIEW

**Comment by JJRR5** [comment]
@sebasdrk17 could you please check?

File: open_academy/models/course.py
Comment (sebasdrk17) [inline open_academy/models/course.py]: Screen Shot 2022-08-29 at 7 36 17 p m

Unless you want to specifically convert what's inside `default`

```suggestion
        default = default or {}
```

File: open_academy/models/course.py
Comment (sebasdrk17) [inline open_academy/models/course.py]:
```suggestion
        copied_count = self.search_count([('name', '=like', 'Copy of %s') %s self.name])
```

Comment (sebasdrk17) [reply open_academy/models/course.py]: Do the same for all the other `format` lines please

File: open_academy/models/course.py
Comment (sebasdrk17) [inline open_academy/models/course.py]:
```suggestion
        default.update({'name': new_name})
```

File: open_academy/models/session.py
Comment (sebasdrk17) [inline open_academy/models/session.py]: Could you try something like this?

this will give you somehitng like:

`res.partner(1) in res.partner(1,2,3,4......)`

```suggestion
        if self.instructor_id in self.atendees_ids:
                raise ValidationError("An instructor cannot be an attendee of his own session")
```

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
made comments

**Comment by JJRR5** [comment]
> made comments

Changes done, could you please check @sebasdrk17 ?

File: open_academy/views/course_views.xml
Comment (sebasdrk17) [inline open_academy/views/course_views.xml]: remove space

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
made comments

**Comment by JJRR5** [comment]
> made comments

Changes done @sebasdrk17 could you please check?

**Comment by sebasdrk17** [comment]
squash and wait for andrea's approval

@andreaarce could you review please?

File: open_academy/models/course.py
Comment (andreaarce) [inline open_academy/models/course.py]: You can skip the evaluation of the else with something like 

```suggestion python 
        new_name =  'Copy of %s' % self.name
        new_name += '(%s)' % copied_count if copied_count > 0 else ''
```
Please try and test this

File: open_academy/models/course.py
Comment (andreaarce) [inline open_academy/models/course.py]:
```suggestion
        return super().copy(default=default)
```

File: open_academy/models/session.py
Comment (andreaarce) [inline open_academy/models/session.py]: Here is missing a for because you can get one o multiple records

```suggestion
        for record in self:
            if record.instructor_id in record.attendees_ids:
```

File: open_academy/models/course.py
Comment (JJRR5) [reply open_academy/models/course.py]: Thanks for the suggestion @andreaarce I really like this approach

**Comment by JJRR5** [comment]
Changes tested and done @andreaarce could you please check?

**Review by andreaarce — APPROVED** [review]

**Comment by andreaarce** [comment]
LGTM @sebasdrk17

**Review by sebasdrk17 — APPROVED** [review]

**Pull request merged**
